reader.go

```diff
@@ -0,0 +1,79 @@
+package sstable
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+	"log"
+)
+
+// Reader loads the index of an SSTable file into the memory.  If the
+// file doesn't have an index, it scans the file and builds the index.
+// After that, we can query key and values of a key.
+type Reader struct {
+	io.ReadSeeker
+	index map[string][]int64
+}
+
+func Open(r io.ReadSeeker) (*Reader, error) {
+	ss := &Reader{
+		ReadSeeker: r,
+		index:      make(map[string][]int64)}
+	if e := ss.loadIndex(); e != nil {
+		return nil, e
+	}
+	return ss, nil
+}
+
+func must(e error) {
+	if e != nil {
+		log.Fatalf("Fatal due to %v", e)
+	}
+}
+
+func (ss *Reader) hasIndex() (uint64, error) {
+	fileLength, e := ss.Seek(0, io.SeekEnd)
+	must(e)
+
+	if fileLength < indexEndFlagSize {
+		return 0, fmt.Errorf("Too short to have index")
+	}
+
+	_, e = ss.Seek(-indexEndFlagSize, io.SeekEnd)
+	must(e)
+	sp, e := readUint32(ss)
+	if e != nil {
+		return 0, fmt.Errorf("Failed to seek index-end flag: %v", e)
+	}
+	if sp != separator {
```

Comment:
Even if `sp == separator` does not prevent the case that the file only has records and does not have an index. Since the content of record can be anything.
If we want to specify if the file contains index or not, maybe we need a flag **always present** at the beginning (or end) of the file.